src/tezWrapper.ml

```diff
@@ -87,16 +86,19 @@ type tez_wrapper_params =
   (* Internal entrypoints *)
   | Call_vault_receive_tez of (Ligo.address * Ligo.tez)
   | Call_vault_send_tez_to_contract of (Ligo.address * Ligo.tez * Ligo.address)
+  | Call_vault_send_tez_to_vault of (Ligo.address * Ligo.tez * Ligo.address)
   | Call_vault_set_delegate of (Ligo.address * Ligo.key_hash option)
 
+type vault_found = VaultFound | VaultNotFound
+
 (** Find the address of the vault of given user, or originate it on the fly,
   * with Tezos.self_address as the owner. *)
-let[@inline] find_vault_address (state: tez_wrapper_state) (user: Ligo.address) : LigoOp.operation option * tez_wrapper_state * Ligo.address =
-  match Ligo.Big_map.find_opt user state.vaults with
-  | Some vault_address -> (None: LigoOp.operation option), state, vault_address
+let[@inline] find_vault_address_append (vaults: vault_map) (user: Ligo.address) (ops: LigoOp.operation list) : vault_found * LigoOp.operation list * vault_map * Ligo.address =
+  match Ligo.Big_map.find_opt user vaults with
+  | Some vault_address -> VaultFound, ops, vaults, vault_address
   | None ->
     let op, vault_address = originate_vault !Ligo.Tezos.self_address in
-    Some op, {state with vaults = Ligo.Big_map.update user (Some vault_address) state.vaults}, vault_address
+    VaultNotFound, (op :: ops), (Ligo.Big_map.update user (Some vault_address) vaults), vault_address
```

Comment:
This is not an issue for this PR, but more out of curiosity - I wonder if there is a performance (gas cost) overhead associated with using a custom `vault_found` type here versus a boolean.

Reply:
Ah, that's a good question. There is no extra gas cost, since both types are represented in Michelson something like `or unit unit`. It's actually pretty nice, being able to use a domain specific type here without any runtime cost :smile: